storage.bs

```diff
@@ -94,37 +100,252 @@ function retrieveNextChunk(nextChunkInfo) {
+
+
+<h3 id=buckets oldids=boxes>Storage buckets</h3>
+
+<p>A <dfn>storage bucket</dfn> is a place for <a>storage endpoints</a> to store data. Whenever a
+<a>storage bucket</a> is cleared by the user agent, it must be cleared in its entirety.
+<!-- This is probably better defined in some kind of storage lifetime section. -->
 
-<p><a>Storage</a> consists of zero or more <dfn export id=site-storage-unit>storage units</dfn>.
+<p>A <a>storage bucket</a> has a <dfn>bottle map</dfn> of <a>storage identifiers</a> to
+<a>storage bottles</a>.
 
-<p>Each <a for=/>origin</a> has an associated <a>storage unit</a>. A <a>storage unit</a> contains a
-single <dfn export id=bucket oldids=box>bucket</dfn>. [[HTML]]
+<p>A <dfn>storage bottle</dfn> is a part of a <a>storage bucket</a> carved out for a single
+<a>storage endpoint</a>. A <a>storage bottle</a> has a <dfn for="storage bottle">map</dfn>, which is
+initially an empty <a for=/>map</a>. A <a>storage bottle</a> also has a
+<dfn for="storage bottle">proxy map reference set</dfn>, which is initially an empty
+<a for=/>set</a>.
 
+<p>A <a>storage bottle</a>'s <a for="storage bottle">map</a> is where the actual data meant to be
+stored lives. User agents are expected to store this data, and make it available across <a>agent</a>
+and even <a>agent cluster</a> boundaries, in an <a>implementation-defined</a> manner, so that this
+standard and standards using this standard can access the contents.
 
-<h3 id=buckets oldids=boxes>Buckets</h3>
+<hr>
 
-<p>A <a>bucket</a> has <dfn export for=bucket oldids=box-mode>mode</dfn> which is either
-"<code title>best-effort</code>" or "<code title>persistent</code>". A
-<dfn export oldids=persistent-box>persistent bucket</dfn> is a <a>bucket</a> whose
-<a for=bucket>mode</a> is "<code title>persistent</code>". A
-<dfn export oldids=non-persistent-box>non-persistent bucket</dfn> is a <a>bucket</a> whose
-<a for=bucket>mode</a> is <em>not</em> "<code title>persistent</code>".
+<p>A <dfn id=bucket oldids=box>more-durable storage bucket</dfn> is a <a>storage bucket</a> for
+more-durable storage APIs.
 
-<p>A <a>bucket</a> is considered to be an atomic unit. Whenever a <a>bucket</a> is cleared by the
-user agent, it must be cleared in its entirety.
+<p>A <a>more-durable storage bucket</a> has a
+<dfn for="more-durable storage bucket" id=bucket-mode oldids=box-mode>mode</dfn>, which is
+"<code>best-effort</code>" or "<code>persistent</code>". It is initially "<code>best-effort</code>".
+
+<hr>
+
+<p>A <dfn>session storage bucket</dfn> is a <a>storage bucket</a> for session storage APIs.
+
+<hr>
+
+<p>To <dfn>create a storage bucket</dfn>, given a <a>storage type</a> <var>type</var>, run these
+steps:
+
+<ol>
+ <li><p>Let <var>bucket</var> be null.
+
+ <li><p>If <var>type</var> is "<code>more-durable</code>", then set <var>bucket</var> to a new
+ <a>more-durable storage bucket</a>.
+
+ <li><p>Otherwise, if <var>type</var> is "<code>session</code>", set <var>bucket</var> to a new
```

Comment:
nit: add "then" (I think we try and use if/then consistently?)

Reply:
We don't use it after otherwise.